refactor(forms): convert Signal Forms errors to use RuntimeError

kirjs opened the pull request:
Signal Forms was using plain `Error` objects instead of Angular's `RuntimeError` class. This PR converts all 13 error throw sites across Signal Forms to use `RuntimeError` with proper error codes in the 1900-1999 range.

JeanMeche reviewed (approved):
reviewed-for: public-api

kirjs committed:
refactor(forms): convert Signal Forms errors to use RuntimeError

- Added 13 new error codes to forms/src/errors.ts (1900-1999)
- use RuntimeError
